### PR TITLE
Added release notification script

### PR DESCRIPTION
**Pull request by kschumy**
*Issue #, if available:*
N/A

*Description of changes:*
Adds a script to publish a notification about a release to an SNS topic.

Intended to work with these changes https://github.com/aws/eks-distro/pull/234

Requires this PR to be able to run for 1-19: https://github.com/aws/eks-distro/pull/235

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: release/sns_notification.sh
Comment (kschumy) [inline release/sns_notification.sh:75]: Requiring "publish" instead of "y" was intended to be cumbersome. Publishing to an SNS topic cannot be undone, and sending incorrect information or spamming the wrong topic results in a bad experience for the subscribers. The script creates all the values used in the notification, and this confirmation step is the only chance the user has to view and check if the ARN and notification content are correct. Cumbersome-ing up the process signals to users that they shouldn't just reflexively or absentmindedly enter "y".

We do not intent to include this script in the automation process. If the user accidentally enters "y" instead of "publish", re-running the scripts takes like a second. As long as the user doesn't confirm publishing, there is no result or impact in running the script over and over again.

**Comment by TerryHowe** [comment]
I was hoping to have the prow job ready before the actual announcement merged

**Comment by kschumy** [comment]
> I was hoping to have the prow job ready before the actual announcement merged

Ok!

File: release/sns_notification.sh
Comment (TerryHowe) [inline release/sns_notification.sh:67]: We are going to use prow (I hope) to push these notifications so, while this is technically correct here, the PR merge is the confirmation we are doing the right thing.

File: release/1-19/release-announcement.txt
Comment (TerryHowe) [inline release/1-19/release-announcement.txt:1]: If you put this announcement in another PR, that will simplify things. Also we should have a url in the announcement probably so people have something to give them more context like a link to https://distro.eks.amazonaws.com/releases/1-19/1/ for this one

File: release/sns_notification.sh
Comment (TerryHowe) [inline release/sns_notification.sh:33]:
```suggestion
```

This isn't customer facing, so the less documentation the better. This will get out of date and confuse the next person. The code is the docs.

File: release/sns_notification.sh
Comment (TerryHowe) [inline release/sns_notification.sh:35]: Prow is going to be triggered by a file change, probably `release/*/release-announcement.txt` it would be a pain to parse that file name to extract the release branch so it could be recreated here. This first argument should be ANNOUNCEMNT_FILE or something of the sort.

File: release/sns_notification.sh
Comment (TerryHowe) [inline release/sns_notification.sh:36]:
```suggestion
SNS_TOPIC_ARN="${2?Second required argument is SNS topic ARN}"
```

**Comment by kschumy** [comment]
/hold

splitting this up into two prs

File: release/sns_notification.sh
Comment (kschumy) [reply release/sns_notification.sh:75]: Never mind. As described in other comments, we're going in a different direction

File: release/sns_notification.sh
Comment (kschumy) [reply release/sns_notification.sh:81]: I'm getting rid of all the confirmation stuff, but is it ok if we print the message info? It could be helpful in verifying what (should be) published without making us track it down via email, sqs, etc.

Comment (TerryHowe) [reply release/sns_notification.sh:81]: Yeh, sounds good

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**
